### ISSUE TITLE
Pkg.installed fails when packages are removed but Project.toml is not yet commited

### ISSUE THREAD

**Issue by christopher-dG** [JuliaLang/Pkg.jl#1332]
Generate a project, along with a Git repo, and add a package:

```jl
(v1.2) pkg> generate Foo
Generating project Foo:
    Foo/Project.toml
    Foo/src/Foo.jl

shell> git -C Foo init
Initialized empty Git repository in /home/degraafc/scratch/Foo/.git/

shell> git -C Foo commit -m "!" --allow-empty
[master (root-commit) a7358cc] !

(v1.2) pkg> activate Foo
Activating environment at `~/scratch/Foo/Project.toml`

(Foo) pkg> add Example
  Updating registry at `~/.local/share/julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed Example ─ v0.5.3
  Updating `~/scratch/Foo/Project.toml`
  [7876af07] + Example v0.5.3
  Updating `~/scratch/Foo/Manifest.toml`
  [7876af07] + Example v0.5.3

shell> git -C Foo add .

shell> git -C Foo commit -m "!"
[master 420e141] !
 3 files changed, 18 insertions(+)
 create mode 100644 Manifest.toml
 create mode 100644 Project.toml
 create mode 100644 src/Foo.jl

julia> using Pkg

julia> Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 1 entry:
  "Example" => v"0.5.3"
```

Now see what happens if we remove it but don't commit yet:

```jl
(Foo) pkg> rm Example
  Updating `~/scratch/Foo/Project.toml`
  [7876af07] - Example v0.5.3
  Updating `~/scratch/Foo/Manifest.toml`
  [7876af07] - Example v0.5.3

julia> Pkg.installed()
ERROR: type Nothing has no field ver
Stacktrace:
 [1] getproperty(::Any, ::Symbol) at ./Base.jl:20
 [2] __installed(::Pkg.Types.PackageMode) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:255
 [3] installed() at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:249
 [4] top-level scope at REPL[17]:1
```

But if we commit, the error goes away:

```jl
shell> git -C Foo commit -a -m "!"
[master 13b76cb] !
 2 files changed, 7 deletions(-)

julia> Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 0 entries
```

If we get ourselves back into the same state and remove the Git repo, the error doesn't show up:

```jl
shell> git -C Foo reset HEAD^
HEAD is now at d800183 !

shell> rm -rf Foo/.git

julia> Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 0 entries
```

**Comment by 00vareladavid** [comment]
`Pkg.installed` went through the display code, which is probably why git's state even matters.

`Pkg.installed` was a vestigial API from Pkg2, but it was never fully fleshed out. I have replaced it with [Pkg.dependencies](https://julialang.github.io/Pkg.jl/dev/api/#Pkg.dependencies). It is more correct and can handle more complicated questions. I just checked it against your example, and it does the right thing.
